File: internal/zerg/zerg.go
package zerg

import "fmt"

// Themed error messages
const (
	MsgDroneNotFound    = "Your warriors have engaged the enemy. ...just kidding. Drone not found."
	MsgDroneExists      = "Evolution chamber occupied."
	MsgSpawnStarting    = "Spawning drone..."
	MsgSpawnComplete    = "Spawning drone... \U0001f7e2 Hatchery ready."
	MsgSlayComplete     = "Drone eliminated. The swarm grows weaker."
	MsgBurrowComplete   = "Drone burrowed. Awaiting further orders."
	MsgUnburrowComplete = "Drone unburrowed. Ready for combat."
	MsgNoDrones         = "The hive is empty. Spawn some drones."
	MsgRepoNotFound     = "We require more minerals."
	MsgTooManyDrones    = "Spawn more overlords. Insufficient vespene gas."
	MsgHighMemory       = "Nuclear launch detected."
	MsgCloning          = "Assimilating genetic material..."
	MsgWaitingTailscale = "Awaiting neural link..."
	MsgDockerError      = "Creep tumor malfunction."
	MsgSocketCreated    = "Extractor online."
	MsgSocketRemoved    = "Extractor offline."
	MsgTokenRefreshed   = "Essence absorbed."
	MsgRecovering       = "Rebuilding creep network..."
)

// Status formats a drone status line for list output.
func Status(name, state string) string {
	icon := "\U0001f534" // red
	switch state {
	case "running":
		icon = "\U0001f7e2" // green
	case "exited":
		icon = "\u26aa" // white
	}
	return fmt.Sprintf("  %s %s [%s]", icon, name, state)
}

// Errorf wraps a message in zerg flavor.
func Errorf(msg string, args ...any) error {
	return fmt.Errorf(msg, args...)
}

// Printf prints a themed message to stdout.
func Printf(msg string, args ...any) {
	fmt.Printf(msg+"\n", args...)
}

File: src/zerg.ts
export class HatcheryError extends Error {
  constructor(message: string) {
    super(message);
    this.name = "HatcheryError";
  }
}

export const msg = {
  droneNotFound:
    "Your warriors have engaged the enemy. ...just kidding. Drone not found.",
  droneExists:
    "A drone for this repo is already running. Slay it first, or use unburrow.\n  Evolution chamber occupied — one larva per cocoon, Overmind.",
  spawnStarting: "Spawning drone...",
  spawnComplete: "Spawning drone... 🟢 Hatchery ready.",
  slayComplete: "Drone eliminated. The swarm grows weaker.",
  burrowComplete: "Drone burrowed. Awaiting further orders.",
  unburrowComplete: "Drone unburrowed. Ready for combat.",
  noDrones: "The hive is empty. Spawn some drones.",
  repoNotFound: "We require more minerals.",
  tooManyDrones: "Spawn more overlords. Insufficient vespene gas.",
  highMemory: "Nuclear launch detected.",
  cloning: "Assimilating genetic material...",
  dockerError: "Creep tumor malfunction.",
  socketCreated: "Extractor online.",
  socketRemoved: "Extractor offline.",
  tokenRefreshed: "Essence absorbed.",
  recovering: "Rebuilding creep network...",
  spawnTimeout:
    "Container did not start within 2 minutes. Check Docker logs for details.\n  Drone lost in the Nydus Network — spawning sequence timed out.",
  updating: "Absorbing latest mutations...",
  updateFailed: "Pull failed — spawning with existing genetic material. Adapt or perish.",
  noEvolutionPlan:
    "No devcontainer.json found in the repo. Add one so the drone knows how to build.\n  The spawning pool rejects unsequenced DNA — evolution requires a build plan.",
  repoUpdated: "Genetic sequence updated.",
  repoConnected: "Neural link established. Drone assimilating new DNA.",
  repoDisconnected: "Genetic strand severed. Drone genome simplified.",
  repoAlreadyConnected: "This genome is already part of the drone's DNA.",
  repoNotConnected: "Cannot sever what was never linked.",
  repoCannotRemovePrimary: "Cannot disconnect the primary genome — the drone would lose its identity.",
};

export function status(name: string, state: string): string {
  const icon =
    state === "running" ? "🟢" : state === "exited" ? "⚪" : "🔴";
  return `  ${icon} ${name} [${state}]`;
}
